src/main/scala/satify/update/solver/Solver.scala

```diff
@@ -51,17 +57,17 @@ object Solver:
   /** Private implementation of [[Solver]] */
   private case class DpllSolver(converter: Converter) extends Solver:
 
-    import DpllSolverMemoize.runDpll
+    override def solveAll(cnf: CNF): Solution = enumerate(cnf)
 
-    override def solve(cnf: CNF): Solution = runDpll(cnf)
+    override def solve(cnf: CNF): Solution = DpllFinder.dpll(cnf)
 
     override def solve(exp: Expression): Solution = solve(converter.convert(exp))
```

Comment:
- `solveAll` name maybe is not really clear, we'll search for a better name
- We need another overloaded method to take in input also an `Expression` (like normal `solve`)

Reply:
Why not?
I'm going to make the second change, thanks for the clarification.